src/Transports.AspNetCore/Extensions/GraphQLHttpEndpointRouteBuilderExtensions.cs

```diff
@@ -11,51 +11,61 @@ namespace Microsoft.AspNetCore.Builder;
-            throw new ArgumentNullException(nameof(endpoints));
+        var opts = new GraphQLHttpMiddlewareOptions();
+        configureMiddleware?.Invoke(opts);
 
-        var requestDelegate = endpoints.CreateApplicationBuilder().UseMiddleware<GraphQLHttpMiddleware<TSchema>>().Build();
-        return new GraphQLHttpEndpointConventionBuilder(endpoints.Map(pattern, requestDelegate).WithDisplayName("GraphQL"));
+        var requestDelegate = endpoints.CreateApplicationBuilder().UseMiddleware<GraphQLHttpMiddleware<TSchema>>(opts).Build();
+        return new GraphQLEndpointConventionBuilder(endpoints.Map(pattern, requestDelegate).WithDisplayName("GraphQL"));
     }
 
     /// <summary>
-    /// Add the GraphQL middleware to the HTTP request pipeline
+    /// Add the GraphQL middleware to the HTTP request pipeline for the specified schema.
     /// </summary>
-    /// <typeparam name="TSchema">The implementation of <see cref="ISchema"/> to use</typeparam>
     /// <typeparam name="TMiddleware">Custom middleware inherited from <see cref="GraphQLHttpMiddleware{TSchema}"/></typeparam>
     /// <param name="endpoints">Defines a contract for a route builder in an application. A route builder specifies the routes for an application.</param>
     /// <param name="pattern">The route pattern.</param>
+    /// <param name="args">The arguments to pass to the middleware type instance's constructor.</param>
     /// <returns>The <see cref="IApplicationBuilder"/> received as parameter</returns>
-    public static GraphQLHttpEndpointConventionBuilder MapGraphQL<TSchema, TMiddleware>(this IEndpointRouteBuilder endpoints, string pattern = "graphql")
-         where TSchema : ISchema
-         where TMiddleware : GraphQLHttpMiddleware<TSchema>
+    public static GraphQLEndpointConventionBuilder MapGraphQL<TMiddleware>(this IEndpointRouteBuilder endpoints, string pattern = "graphql", params object[] args)
+        where TMiddleware : GraphQLHttpMiddleware
     {
-        if (endpoints == null)
-            throw new ArgumentNullException(nameof(endpoints));
-
-        var requestDelegate = endpoints.CreateApplicationBuilder().UseMiddleware<TMiddleware>().Build();
-        return new GraphQLHttpEndpointConventionBuilder(endpoints.Map(pattern, requestDelegate).WithDisplayName("GraphQL"));
+        var requestDelegate = endpoints.CreateApplicationBuilder().UseMiddleware<TMiddleware>(args).Build();
+        return new GraphQLEndpointConventionBuilder(endpoints.Map(pattern, requestDelegate).WithDisplayName("GraphQL"));
     }
 }
 
 /// <summary>
 /// Builds conventions that will be used for customization of Microsoft.AspNetCore.Builder.EndpointBuilder instances.
 /// Special convention builder that allows you to write specific extension methods for ASP.NET Core routing subsystem.
 /// </summary>
-public class GraphQLHttpEndpointConventionBuilder : IEndpointConventionBuilder
+public class GraphQLEndpointConventionBuilder : IEndpointConventionBuilder
```

Comment:
Remove Http?

Reply:
To match the name of all the rest of similar ASP.NET Core classes, such as:
- `ControllerActionEndpointConventionBuilder`
- `ComponentEndpointConventionBuilder`
- `HubEndpointConventionBuilder`

Not a big deal; the whole class could be removed as far as I'm concerned.  I figured perhaps sometime in the future we would actually do something with it, so I left it there.

Reply:
As a side note, not all MS extension methods return a custom SomethingConventionBuilder class.  Some just return `IEndpointConventionBuilder`.  We could do the same, and reduce the API here.

Reply:
Understood. OK to keep `GraphQLEndpointConventionBuilder`.